cmd/aws-cloud-controller-manager/main.go

```diff
@@ -107,28 +110,74 @@ func main() {
 				}
 			}
 
-			printFlags(cmd.Flags())
+			cliflag.PrintFlags(cmd.Flags())
 
-			c, err := s.Config(KnownControllers(), ControllersDisabledByDefault.List())
+			c, err := s.Config(KnownControllers(), app.ControllersDisabledByDefault.List())
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "%v\n", err)
 				os.Exit(1)
 			}
 
-			if err := app.Run(c.Complete(), wait.NeverStop); err != nil {
+			// initialize cloud provider with the cloud provider name and config file provided
+			cloud, err := cloudprovider.InitCloudProvider(cloudProvider, "")
+			if err != nil {
+				klog.Fatalf("Cloud provider could not be initialized: %v", err)
+			}
+			if cloud == nil {
+				klog.Fatalf("Cloud provider is nil")
+			}
+
+			if !cloud.HasClusterID() {
+				if c.ComponentConfig.KubeCloudShared.AllowUntaggedCloud {
+					klog.Warning("detected a cluster without a ClusterID.  A ClusterID will be required in the future.  Please tag your cluster to avoid any future issues")
+				} else {
+					klog.Fatalf("no ClusterID found.  A ClusterID is required for the cloud provider to function properly.  This check can be bypassed by setting the allow-untagged-cloud option")
+				}
+			}
+
+			// Initialize the cloud provider with a reference to the clientBuilder
+			cloud.Initialize(c.ClientBuilder, make(chan struct{}))
+			// Set the informer on the user cloud object
+			if informerUserCloud, ok := cloud.(cloudprovider.InformerUser); ok {
+				informerUserCloud.SetInformers(c.SharedInformers)
+			}
+
+			controllerInitializers = app.DefaultControllerInitializers(c.Complete(), cloud)
+
+			if err := app.Run(c.Complete(), controllerInitializers, wait.NeverStop); err != nil {
 				fmt.Fprintf(os.Stderr, "%v\n", err)
 				os.Exit(1)
 			}
 		},
+		Args: func(cmd *cobra.Command, args []string) error {
+			for _, arg := range args {
+				if len(arg) > 0 {
+					return fmt.Errorf("%q does not take any arguments, got %q", cmd.CommandPath(), args)
+				}
+			}
+			return nil
+		},
 	}
 
 	fs := command.Flags()
-	namedFlagSets := s.Flags(KnownControllers(), ControllersDisabledByDefault.List())
+	namedFlagSets := s.Flags(app.KnownControllers(controllerInitializers), app.ControllersDisabledByDefault.List())
+	verflag.AddFlags(namedFlagSets.FlagSet("global"))
 	globalflag.AddGlobalFlags(namedFlagSets.FlagSet("global"), command.Name())
 
 	for _, f := range namedFlagSets.FlagSets {
 		fs.AddFlagSet(f)
 	}
+	usageFmt := "Usage:\n  %s\n"
+	cols, _, _ := term.TerminalSize(command.OutOrStdout())
+	command.SetUsageFunc(func(cmd *cobra.Command) error {
```

Comment:
@andrewsykim I referred to https://github.com/kubernetes/cloud-provider/blob/master/sample/basic_main.go but I'm not sure if we need this for now 🤔